test/test_output_writers.jl

```diff
@@ -35,11 +35,65 @@ function run_thermal_bubble_netcdf_tests()
     @test all(S .≈ data(model.tracers.S))
 end
 
+"""
+Run two coarse rising thermal bubble simulations and make sure that when
+restarting from a checkpoint, the restarted simulation matches the non-restarted
+simulation numerically.
+"""
+function run_thermal_bubble_checkpointer_tests(arch)
+    Nx, Ny, Nz = 16, 16, 16
+    Lx, Ly, Lz = 100, 100, 100
+    Δt = 6
+
+    true_model = Model(N=(Nx, Ny, Nz), L=(Lx, Ly, Lz), ν=4e-2, κ=4e-2, arch=arch)
+
+    # Add a cube-shaped warm temperature anomaly that takes up the middle 50%
+    # of the domain volume.
+    i1, i2 = round(Int, Nx/4), round(Int, 3Nx/4)
+    j1, j2 = round(Int, Ny/4), round(Int, 3Ny/4)
+    k1, k2 = round(Int, Nz/4), round(Int, 3Nz/4)
+    true_model.tracers.T.data[i1:i2, j1:j2, k1:k2] .+= 0.01
+
+    checkpointed_model = deepcopy(true_model)

```

Comment:
Why is it necessary to copy the model before checkpointing?

Reply:
It's creating a second model that will be checkpointed as opposed to `true_model` which isn't.

Probably over paranoid but I wanted the two models to be time-stepped separately.

Reply:
Are you worried that checkpointing will modify `model`?

Since this behavior is undesirable / unexpected, perhaps it should simply be tested for, so that we can assume that the model is not modified.